python/mxnet/ndarray/sparse.py

```diff
@@ -830,30 +957,65 @@ def row_sparse_array(data, indices, shape, ctx=None, dtype=None, indices_type=No
 
     Example
     -------
-    >>> a = mx.nd.sparse.row_sparse_array([[1, 2], [3, 4]], [1, 4], (6, 2))
+    >>> a = mx.nd.sparse.row_sparse_array(([[1, 2], [3, 4]], [1, 4]), shape=(6, 2))
     >>> a.asnumpy()
     array([[ 0.,  0.],
            [ 1.,  2.],
            [ 0.,  0.],
            [ 0.,  0.],
            [ 3.,  4.],
            [ 0.,  0.]], dtype=float32)
+
+    See Also
+    --------
+    RowSparseNDArray : MXNet NDArray in row sparse format.
     """
+    # construct a row sparse array from (D0, D1 ..) or (data, indices)
+    if isinstance(arg1, tuple):
```

Comment:
list and tuple?

Reply:
this is intentional. otherwise there're too many cases since data and indices can be list, too